### PR TITLE
[LSP] Replace logging with tracing

### PR DESCRIPTION
**Pull request by sdankel**
Related https://github.com/FuelLabs/sway/issues/2286

Replacing the tower_lsp logging with calls to the `tracing` lib.

Added an option to `init_tracing_subscriber` to allow all tracing logs to be piped through stderr instead of split between stderr/stdio. Because the main thread in the LSP uses stdio to communicate with the LSP client, tracing logs must go through stderr.

image

### CODE REVIEW

**Comment by JoshuaBatty** [comment]
Nice, this is going to be super handy to have! Looks like there are just a few clippy warnings that need addressing.

File: sway-lsp/src/lib.rs [hunk 1]
```diff
@@ -9,8 +11,16 @@ mod server;
 pub mod test_utils;
 pub mod utils;
 use server::Backend;
+use tracing::metadata::LevelFilter;
 
 pub async fn start() {
+    let tracing_options = TracingSubscriberOptions {
+        log_level: Some(LevelFilter::DEBUG), // TODO: Set this based on IDE config
```
Comment (JoshuaBatty) [inline sway-lsp/src/lib.rs:18]: sounds good, we can have a chat next week and I can talk you though how to go about setting something like this up.

File: sway-lsp/src/server.rs [hunk 2]
```diff
@@ -186,10 +178,8 @@ impl Backend {
 #[tower_lsp::async_trait]
 impl LanguageServer for Backend {
     async fn initialize(&self, params: InitializeParams) -> jsonrpc::Result<InitializeResult> {
-        eprintln!("InitializeParams = {:#?}", params);
-        self.client
-            .log_message(MessageType::INFO, "Initializing the Sway Language Server")
-            .await;
+        tracing::error!("InitializeParams = {:#?}", params);
```
Comment (JoshuaBatty) [inline sway-lsp/src/server.rs:181]: The existing `eprintln!("InitializeParams = {:#?}", params);` was accidentally left over from a previous PR. Can we remove this line?

**Review by JoshuaBatty — CHANGES_REQUESTED** [review]
LGTM, thanks for this! Just one thing to be removed and I'll be happy to approve.

**Review by JoshuaBatty — DISMISSED** [review]

**Review by mitchmindtree — APPROVED** [review]

**Review by otrho — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**